Attempt to prevent information disclosure

alexandreelise opened the pull request:
For example if someone is looking over your shoulder while you have backup oteps shown. This little modification tries to mitigate this possibility.
One side note. The details text should be a language string.

Pull Request for Issue #37969.

### Summary of Changes

add details/summary html5 tags

### Testing Instructions

Enable 2FA in administrator for a user in Registered group
Login in frontend with this Registered user
Go to edit profile
Scroll down to see Oteps backup codes

### Actual result BEFORE applying this Pull Request

Oteps backup codes show immediately and might disclose information

### Expected result AFTER applying this Pull Request

Oteps Backup codes don't show immediately. Rather, they show only if we click on Details

### Documentation Changes Required
Possibly.

Commit:
Attempt to prevent information disclosure

For example if someone is looking over your shoulder while you have backup oteps shown. This little modification tries to mitigate this possibility.
One side note. The details text should be a language string.

brianteeman commented on components/com_users/tmpl/profile/edit.php:
```diff
@@ -100,9 +100,12 @@
 					</div>
 				<?php else : ?>
+					<details>
+						<summary>Details</summary>
```
do not hard code english. use a language string

alexandreelise replied:
Yes you are right @brianteeman I put it in a side note as I know it is best practice to use language strings but I wasn't sure what language string to add and I the translation team had a better idea for the language string name that's why I hardcoded the English. Next time I won't do the same mistake.

richard67 replied:
That’s not a reason not to do it right here. If you don’t modify your PR so it uses a language string here, it can’t be accepted.

Commit:
[4.1] Add language string for oteps details

wojsmol commented on language/en-GB/com_users.ini:
```diff
@@ -51,6 +51,7 @@ COM_USERS_PROFILE_NAME_LABEL="Name"
 COM_USERS_PROFILE_NEVER_VISITED="This is the first time you have visited this site."
 COM_USERS_PROFILE_NOCHANGE_USERNAME_DESC="If you want to change your username, please contact a site administrator."
 COM_USERS_PROFILE_OTEPS="One time emergency passwords"
+COM_USERS_PROFILE_OTEPS_DETAILS_DESC="Details"
```
Use COM_USERS_PROFILE_OTEPS_DETAILS_LABEL="Details"

alexandreelise replied:
Thanks for the suggestion @wojsmol . It has been fixed

Commit:
[4.1] Add language string to improve multilingual

Require language strings for best practices and for this PR #37970 to be accepted as required by the project maintainers

Commit:
[4.1] Edit language string for oteps multilingual

Edit language string as suggested by project maintainers

Commit:
[4.1] Edit language string for oteps details

richard67 commented:
@alexandreelise Thanks for this PR. Unfortunately I am not sure if it will make sense in 4.1 because there will be only one more 4.1.x release before 4.2.0 will come, and in 4.2. the 2FA will be completely reworked and renamed to MFA (multi-factor authentication). See PR #37912 for the ongoing work.

I haven't checked yet if that PR will bring back the thing which you want to solve with this PR here. But that might be the case, and your changes would be lost.

@brianteeman As you have checked Nik's PR already: Do you know if it already works like that, show the backup codes only on demand?

brianteeman commented:
if there is even just one more release then this should be included even if the replacement code in 4.2 will not use it

Of course if you use the global string of JDETAILS then there is no problem at all.

Commit:
[4.1] Remove useless language string

Now using JDETAILS as it already exists as suggested by project maintainers

Commit:
[4.1] Using already defined language string

As suggested by a project maintainer using an already defined language string so we don't have to maintain it. Easier to merge.

brianteeman commented:
As there will now not be another 4.1 release this can be closed as it is not relevant for 4.2
